src/main/java/org/owfs/jowfsclient/internal/OwfsConnectionImpl.java

```diff
@@ -74,16 +75,19 @@ private boolean connect(boolean force) throws IOException {
 	}
 
 	private boolean tryToSocketConnectionAndStreamsInitialization() throws IOException {
-		owSocket = new Socket();
+	    owSocket = new Socket();
+	    for (InetAddress inetAddress : InetAddress.getAllByName(config.getHostName())) {
 		try {
-			owSocket.connect(new InetSocketAddress(config.getHostName(), config.getPortNumber()), config.getConnectionTimeout());
-			owIn = new DataInputStream(owSocket.getInputStream());
-			owOut = new DataOutputStream(owSocket.getOutputStream());
-			return true;
```

Comment:
Not really, if the connect fails the exception is caught, then the cleint connects to next inetAdress.
So if first ipv6 2nd ipv4 and server is running on ipv6 the owSocket.connect is only executed once.
ipv6 then ipv4 server on ipv4 owSocket first throws Exception, second attempt it connects